include/libtorrent/disk_observer.hpp

```diff
@@ -37,7 +37,7 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent {
 
-	struct TORRENT_EXTRA_EXPORT disk_observer
+	struct TORRENT_EXPORT disk_observer
```

Comment:
is this really part of the public API? I don't think so, it's just used for peers to be notified when under back-pressure from the disk threads, isn't it?

Reply:
I initially moved it to `aux_/`, but then I found that it's used as part of the public API in `disk_interface`, at https://github.com/arvidn/libtorrent/blob/master/include/libtorrent/disk_interface.hpp#L176

Not sure how do you want to handle this.

Reply:
that's a good point. This will make this type show up in the documentation too.